### PR TITLE
Fix crash from RoomGame refactor

### PR DESCRIPTION
**Pull request by sparkychildcharlie**
RoomGame's constructor currently sets the room's game to whatever roomgame is being created.

This sets ``self`` as the parent game, lets the child game overwrite ``room.game`` and sets it back to the parent game after creating a hunt.

### CODE REVIEW

**Comment by sparkychildcharlie** [comment]
@Zarel I hope this way of fixing is acceptable.  I've explained on discord why I need both things to be a roomgame (to track renames).  Having them both in the same room because one of the 2 serves as a "control device" and not an actual game.

**Comment by Zarel** [comment]
I've explained on Discord that you can't just create a "free-floating" roomgame. All our code expects the roomgame to be `room.game`.

https://github.com/Zarel/Pokemon-Showdown/blob/0c1e1087397187a6af8a3cd6e10a75f8624189db/server/users.ts#L765

For instance, this crash, which I've had to suppress, is nearly definitely caused by your code.

https://github.com/Zarel/Pokemon-Showdown/blob/0c1e1087397187a6af8a3cd6e10a75f8624189db/server/users.ts#L1004

As is this.

File: server/chat-plugins/scavenger-games.js [hunk 1]
```diff
@@ -188,10 +188,16 @@ class ScavGame extends Rooms.RoomGame {
...
 		this.onStartEvent();
-		this.childGame = new Rooms.ScavengerHunt(room, staffHost, host, gameType, questions, this);
+		this.startHunt(room, staffHost, host, gameType, questions, this);
 		return true;
 	}
 
+	startHunt(...args) {
+		let self = this;
+		self.childGame = new Rooms.ScavengerHunt(...args);
+		this.room.game = self;
```
Comment (xfix) [inline server/chat-plugins/scavenger-games.js]: I'm not sure if I like this, seems very hacky to me.

**Comment by Zarel** [comment]
This is still not fixed. Should we just remove the Scavengers plugin?

**Comment by KrisXV** [comment]
No. Absolutely not. It'll be fixed if this gets typescripted and uses room.getGame i presume.

**Comment by Zarel** [comment]
The problem isn't that simple, or I would have fixed it myself.

**Comment by Zarel** [comment]
It's not like `.js` files _can't_ use `room.getGame`

**Comment by Zarel** [comment]
I think the correct interim solution is to make one of these games a `minorActivity`.

**Comment by sparkychildcharlie** [comment]
This is fixed in #6507 simply by making the scavenger game into a non-roomgame activity simply used to keep a list of userids of players who are still in the game

**Pull request closed without merging**